game-core/src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -99,15 +105,24 @@
    * Warning: game engine code invokes this method to spawn new game clients.
    */
   public static void main(final String[] args) {
-    LoggingConfiguration.initialize();
+    Preconditions.checkState(!GraphicsEnvironment.isHeadless(),
+        "UI client launcher invoked from headless environment. This is current prohibited by design to "
+            + "avoid UI rendering errors in the headless environment.");
+
+    ErrorHandler.registerExceptionHandler();
     ClientSetting.initialize();
 
     if (!ClientSetting.USE_EXPERIMENTAL_JAVAFX_UI.booleanValue()) {
-      Interruptibles.await(() -> SwingAction.invokeAndWait(() -> {
-        LookAndFeel.setupLookAndFeel();
-        ErrorConsole.createConsole();
-        ErrorMessage.INSTANCE.init();
-      }));
+      final Console console = new Console();
+      final SimpleFormatter formatter = new SimpleFormatter();
+      LogManager.getLogManager().getLogger("").addHandler(new ConsoleHandler(
+          logMsg -> {
+            ErrorMessage.show(logMsg.getMessage());
+            console.append(formatter.format(logMsg));
```

Comment:
Another option to consider here is to only display the popup when the log level is `ERROR`.  That might make it less contentious to consider logging at lower levels, as those records won't result in a popup.

Reply:
Turns out it's needed when we go below warning, we've got a lot of logging at that level : |

I don't think it should be so contentious to have logging at low levels, just more considered temporary unless we strategically plan to have those data points.